src/Surfnet/AzureMfa/Application/Service/AzureMfaService.php

```diff
@@ -199,6 +202,20 @@ public function handleResponse(Request $request): User
 
         $attributes = $assertion->getAttributes();
 
+        // TODO: This should be a config parameter in institutions.yaml instead of string matching
+        // If the IDP was an AzureAD endpoint (the entityID or Issuer starts with https://login.microsoftonline.com/, or preferably an config parameter in institutions.yaml)
+        // the SAML response attribute 'http://schemas.microsoft.com/claims/authnmethodsreferences'
+        // should contain 'http://schemas.microsoft.com/claims/multipleauthn'
+        if (strpos($azureMfaIdentityProvider->getSsoUrl(), "https://login.microsoftonline.com/") === 0) {
+            $this->logger->info('This is an AzureAD IdP. Validating authnmethodsreferences in the response.');
+            if (!in_array('http://schemas.microsoft.com/claims/multipleauthn', $attributes['http://schemas.microsoft.com/claims/authnmethodsreferences'])) {
+                // TODO: Create a proper AuthnmethodsreferencesMissingException
+                throw new Exception(
+                    'No http://schemas.microsoft.com/claims/multipleauthn in authnmethodsreferences.'
+                );
+            }
+        }
```

Comment:
I'd like to see this logic encapsulated in a helper service. That way the Azure direct feature support is isolated and more easily maintained. This also enables you to easily dependency inject the configurable parameters into the service.

The block you added above (line160..165) can also be moved there. 

The helper class can then be DI'ed into this service.